fieldservice_account_payment/models/account_payment.py

```diff
@@ -36,8 +36,9 @@ def action_view_fsm_orders(self):
     @api.depends('invoice_ids.fsm_order_ids')
     def _compute_fsm_order_ids(self):
         fsm_order_ids = []
-        for invoice in self.invoice_ids:
-            if invoice.fsm_order_ids:
-                fsm_order_ids.extend(invoice.fsm_order_ids.ids)
-        if fsm_order_ids:
-            self.fsm_order_ids = [(6, 0, fsm_order_ids)]
+        for rec in self:
+            for invoice in rec.invoice_ids:
+                if invoice.fsm_order_ids:
+                    fsm_order_ids.extend(invoice.fsm_order_ids.ids)
+            if fsm_order_ids:
+                rec.fsm_order_ids = [(6, 0, fsm_order_ids)]
```

Comment:
The exact same result can be achieved with this simpler code:
```suggestion
            rec.fsm_order_ids = rec.mapped("invoice_ids.fsm_order_ids")
```